src/core.jsx

```diff
@@ -52,6 +53,11 @@ export default (
         return !!print(this.props, this.context)
       }
 
+      if (print === undefined) {
```

Comment:
I think this is better to move this condition at the beginin of `isLoaded` because this is faster to test undefined than testing types